scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -579,7 +579,8 @@ class FossId internal constructor(
         if (response.data?.status in SCAN_STATE_FOR_TRIGGER) {
             logger.info { "Triggering scan as it has not yet been started." }
 
-            val scanResult = service.runScan(config.user, config.apiKey, scanCode, *runOptions)
+            val scanResult =
+                service.runScan(config.user, config.apiKey, scanCode, *runOptions.withAdditionOptions(config))
```

Comment:
Instead of merging array into one, you could also simply pass two arrays to `runScan`. Maybe something like

    val optionsFromConfig = arrayOf(
        "auto_identification_detect_declaration" to config.detectLicenseDeclarations.compareTo(false).toString(),
        "auto_identification_detect_copyright" to config.detectCopyrightStatements.compareTo(false).toString()
    )

    val scanResult = service.runScan(config.user, config.apiKey, scanCode, *runOptions, *optionsFromConfig)